endpoint/endpoint_test.go

```diff
@@ -1,3 +1,5 @@
+// +build integration
```

Comment:
Probably unintended?

`endpoint_test.go` use mocks all the way, so there's no need to use an actual C*/simulacron instance.

Reply:
With out this change the endpoint integration tests no longer run.

Reply:
I'm not sure I follow, on `master` when I run: `go test -v -count=1 ./endpoint`, I get `TestDataEndpoint_Query`, `TestDataEndpoint_Auth`, `TestDataEndpoint_PageSize` and `TestDataEndpoint_AuthNotProvided` being executed, along with `TestEndpoint` that registers the ginkgo specs for the package. As the there are 0 ginkgo specs (all of them I integration tests), it outputs `Will run 0 of 0 specs`. That's expected.

We should register ginkgo specs outside tags and then let each tag (`integration` and `simulated`) define the actual specs.

Reply:
If you build with the `integration` tag then the tests in `endpoint_integration_test.go` do not run because the test function `TestEndpoint` and the before/after suite handlers in `endpoint_test.go` are not compiled in.

Reply:
It's fine I removed. Does everything else look ok?